__tests__/getAllMarkdownFiles.test.ts

```diff
@@ -0,0 +1,75 @@
+// tests/getAllMarkdownFiles.test.ts
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { getAllMarkdownFiles } from '../utils/getAllMarkdownFiles';
+
+describe('getAllMarkdownFiles', () => {
+    let testDir: string;
+    
+    // Creates temporary directory before each test
+    //  tmp/
+    //      getAllMarkdownFiles-*/
+    //          file1.md
+    //          file2.txt
+    //          README.MD
+    //          subdir/
+    //              file3.md
+    //              index.md
+    beforeEach(() => {
+        testDir = fs.mkdtempSync(path.join(os.tmpdir(), 'getAllMarkdownFiles-'));
+
+        fs.writeFileSync(path.join(testDir, 'file1.md'), '# File 1');
+        fs.writeFileSync(path.join(testDir, 'file2.txt'), 'This is not markdown.');
+        fs.writeFileSync(path.join(testDir, 'README.md'), '# README (uppercase extension)');
+
+        const subDir = path.join(testDir, 'subdir');
+        fs.mkdirSync(subDir);
+        fs.writeFileSync(path.join(subDir, 'file3.md'), '# File 3');
+        fs.writeFileSync(path.join(subDir, 'index.md'), '# Index');
+    });
+
+    // Removes temporary directory after each test
+    afterEach(() => {
+        fs.rmSync(testDir, { recursive: true, force: true });
+    });
```

Comment:
```suggestion
    afterEach(() => {
    if (fs.existsSync(testDir)) {
        fs.rmSync(testDir, { recursive: true, force: true });
    }
});
```
This prevents errors if the directory was removed prematurely.

Reply:
You are right. That would have caused test failures unrelated to the actual test logic.